src/transport/TransportMgr.h

```diff
@@ -49,8 +52,24 @@ class TransportMgrDelegate
      *
      * @param source    the source address of the package
      * @param msgBuf    the buffer containing a full CHIP message (except for the optional length field).
+     * @param ctxt      the pointer to additional context on the underlying transport. For TCP, it is a pointer
+     *                  to the underlying connection object.
      */
-    virtual void OnMessageReceived(const Transport::PeerAddress & source, System::PacketBufferHandle && msgBuf) = 0;
+    virtual void OnMessageReceived(const Transport::PeerAddress & source, System::PacketBufferHandle && msgBuf,
+                                   Transport::MessageTransportContext * ctxt = nullptr) = 0;
```

Comment:
This is a change to a public API, right?  Or is this not public API?

Reply:
I believe it is. This is necessary to know which connection the message arrived on. Otherwise, the implicit assumption would be that there is exactly one connection to a given peer.

Reply:
My point is that we might want to think about making this change in a compatible way: have both the new and old signatures, so old API clients which presumably don't care about the context don't get it, and new ones know to override the new signature, not the old one.

And default impl of the new signature calls the old signature.

Reply:
Agreed. I was resorting to the default arg value of nullptr to the connection context to essentially achieve that. For old api clients, the argument should be not present in the calls and default to null always.

Reply:
> For old api clients, the argument should be not present in the calls and default to null always

Old API clients _implement this method_, no?  So this is a compile-time break for them as currently done.

Reply:
Ok. Apologies for the miscommunication. This is not a SDK public API. It is a public API of the SessionManager and this change terminates in the SessionManager itself. So, beyond that, when the message is sent up to the ExchangeMgr, this is not carried.